WooCommerce/Classes/Extensions/String+Helpers.swift

```diff
@@ -68,4 +68,150 @@ extension String {
+            for replaceTarget in replaceTargets {
+                outString = outString.replacingOccurrences(of: replaceTarget, with: ".")
+            }
+            return outString
+        }
+
+        /// Insert a . before and after any non number
+        ///
+        func insertDotsBeforeAndAfterAnyNonNumber(inString: String) -> String {
+            guard inString.count > 1 else {
+                return inString
+            }
+
+            /// Initialize our output with the first character of the input string
+            var outString = String(inString[inString.index(inString.startIndex, offsetBy: 0)])
+
+            /// Loop over the remaining characters in the string, inserting . as needed
+            for index in 1...inString.count - 1 {
+                let characterAtIndex = inString[inString.index(inString.startIndex, offsetBy: index)]
+                let characterBefore = inString[inString.index(inString.startIndex, offsetBy: index - 1)]
+
+                let characterAtIndexIsNumberOrDot = characterAtIndex.isNumber || characterAtIndex == "."
+                let characterBeforeIsNumberOrDot = characterBefore.isNumber || characterBefore == "."
+
+                if !characterAtIndexIsNumberOrDot && characterBefore.isNumber {
+                    outString += "."
+                }
+
+                if characterAtIndex.isNumber && !characterBeforeIsNumberOrDot {
+                    outString += "."
+                }
+
+                outString += String(characterAtIndex)
+            }
+
+            return outString
+        }
+
+        /// Ranked per https://www.php.net/manual/en/function.version-compare.php
+        ///
+        enum ComponentScore: Comparable {
+            case other
+            case dev
+            case alpha
+            case beta
+            case RC
+            case number
+            case patch
+        }
+
+        /// Score each component
+        ///
+        func scoreComponent(stringComponent: String) -> ComponentScore {
+            if stringComponent == "dev" {
+                return .dev
+            }
+            if stringComponent == "alpha" || stringComponent == "a" {
+                return .alpha
+            }
+            if stringComponent == "beta" || stringComponent == "b" {
+                return .beta
+            }
+            if stringComponent == "RC" || stringComponent == "rc" {
+                return .RC
+            }
+            let componentCharacterSet = CharacterSet(charactersIn: stringComponent)
+            if componentCharacterSet.isSubset(of: .decimalDigits) {
+                return .number
+            }
+
+            if stringComponent == "pl" || stringComponent == "p" {
+                return .patch
+            }
+
+            return .other
+        }
+
+        /// Score and compare two string components
+        ///
+        func compareStringComponent(stringComponent1: String, stringComponent2: String) -> ComparisonResult {
```

Comment:
nit: just a suggestion to follow the signature of some iOS comparators ([example](https://developer.apple.com/documentation/swift/string/standardcomparator/3802578-compare))

```suggestion
        func compareStringComponent(_ lhs: String, _ rhs: String) -> ComparisonResult {
```

Reply:
Fixed in https://github.com/woocommerce/woocommerce-ios/pull/5382/commits/998b0d2a776b4427788215fce738ac4c5108c7cc